app/models/katello/host_collection.rb

```diff
@@ -24,20 +24,33 @@ class HostCollection < Katello::Model
                                                      :message => N_("must be a positive integer value.")}
 
     alias_attribute :content_host_limit, :max_content_hosts
-    validate :validate_max_content_hosts
+
+    validate :max_content_hosts_check, :if => ->(host_collection) { host_collection.new_record? || host_collection.max_content_hosts_changed? }
+
+    validate :max_content_host_value_exists
+
+    validate :max_content_hosts_not_exceeded, :on => :create
```

Comment:
I'd drop the empty lines between each of these three lines and put them above `alias_attribute` so they can go with the other validation stuff.